google/cloud/pubsub/samples/subscription_admin_samples.cc

```diff
@@ -50,6 +52,23 @@ CreateSubscriptionAdminCommand(std::string const& name,
                                                            std::move(adapter)};
 }
 
+// Delete all subscriptions created with that include "cloud-cpp-samples".
+// Ignore any failures. If multiple tests are cleaning up subscriptions in
+// parallel, then the delete call might fail.
+void CleanupSubscriptions(
+    google::cloud::pubsub_admin::SubscriptionAdminClient& client,
+    std::string const& project_id) {
+  auto const parent = google::cloud::Project(project_id).FullName();
+  for (auto& subscription : client.ListSubscriptions(parent)) {
+    if (!subscription) continue;
+    if (!absl::StrContains(subscription->name(), "cloud-cpp-samples")) continue;
```

Comment:
Should we include the date in the subscription ID and then only delete subscriptions that are > N days old?

Reply:
It does not tell you how old topics are. I assumed subscriptions were the same.

Reply:
All of the subscriptions created by samples include cloud-cpp-samples, so it will only delete subscriptions created by running samples.

Unless someone else named their subscription that as well.

Which would be unfortunate.

Reply:
We can change this function:

https://github.com/googleapis/google-cloud-cpp/blob/f7ca8a9c7d3dcc59915017e75c6774d7e831ff71/google/cloud/pubsub/samples/pubsub_samples_common.cc#L150-L154

For inspiration, we do this for bigtable resources:

https://github.com/googleapis/google-cloud-cpp/blob/f7ca8a9c7d3dcc59915017e75c6774d7e831ff71/google/cloud/bigtable/testing/random_names.cc#L42-L53

Reply:
We do put the date in the resource name already, but does it matter 🤔

Reply:
> but does it matter

What if two builds are running at the same time.

1. Build A creates a subscription.
2. Build B deletes all subscriptions.
3. Build A tries to use the subscription from 1.

🫨

Reply:
That's fair. I'll fix this PR and the other one to check the date.

Reply:
Done